chainntnfs/tx_notifier.go

```diff
@@ -1156,6 +1105,105 @@ func (n *TxNotifier) DisconnectTip(blockHeight uint32) error {
 	return nil
 }
 
+// updateHints attempts to update the confirm and spend hints for all relevant
+// transactions and outpoints respectively. The disconnect boolean is used to
+// signal whether we are updating hints after a block has been disconnected or
+// not.
+//
+// NOTE: This must be called with the TxNotifier's lock held and after it's
+// height has already been reflected by a block being connected/disconnected.
+func (n *TxNotifier) updateHints(disconnect bool) {
+	// In order to retrieve the corect transactions and outpoints for which
+	// we'll update hints for, we'll need to take into account if a block
+	// was disconnected or not. If it was, then we'll need to look at the
+	// next height as the TxNotifier's height has already reflected the
+	// block being disconnected.
+	confirmSpendHeight := n.currentHeight
```

Comment:
can the height be passed into the method, instead of doing this?

Reply:
Yeah, definitely much cleaner solution. Fixed!